dynamic.c

```diff
@@ -1608,6 +1608,19 @@ bool rarch_environment_cb(unsigned cmd, void *data)
          break;
       }
 
+      case RETRO_ENVIRONMENT_GET_RESOURCE_DIRECTORY:
+      {
+         const struct retro_resource_path *res = (const struct retro_resource_path*)data;
+
+         const char *system_dir = NULL;
+         if (!rarch_environment_cb(RETRO_ENVIRONMENT_GET_SYSTEM_DIRECTORY, (void*)&system_dir) || system_dir == NULL)
```

Comment:
Hm, when I think about, returning false here if path is not found seems a bit flawed,
because returning false from env callback typically means that the feature is not supported.
The path pointer should be written out to NULL instead here if not found.

I'm leaning more towards static storage here to store the buffer, then passing a const char * to that buffer now, because it seems ugly to overwrite a char * that potentially is allocated on heap to solve this.